cmake/Modules/ProtocGenUpb.cmake

```diff
@@ -0,0 +1,389 @@
+# This script is to be used to provide a function to
+# call the protoc executable using the upb plugins.
+#
+# The following variables can be set and are optional:
+#
+# PROTOBUF_SRC_ROOT_FOLDER - When compiling with MSVC, if this cache variable is
+#                            set the protobuf-default VS project build locations
+#                            (vsprojects/Debug & vsprojects/Release) will be
+#                            searched for libraries and binaries.
+#
+# PROTOC_PATH              - Path to protoc executable
+# UPB_PLUGIN_PATH          - Path to protoc-gen-upb executable
+# UPBDEF_PLUGIN_PATH       - Path to protoc-gen-upbdef executable
+#
+# UPB_IMPORT_DIRS          - List of additional directories to be searched for
+#                            imported .proto files.
+#
+# UPB_GENERATE_X_APPEND_PATH    - By default -I will be passed to protoc for
+#                                 each directory where a proto file is referenced.
+#                                 This causes all output files to go directly under
+#                                 build directory, instead of mirroring relative
+#                                 paths of source directories. Set to FALSE
+#                                 if you want to disable this behaviour.
+#
+# Defines the following variables:
+#
+# The following cache variables are also available to set or use:
+# PROTOBUF_PROTOC_EXECUTABLE - The protoc compiler (full path if not found in ${PATH})
+# UPB_GENERATOR     - The upb plugin executable (full path if not found in ${PATH})
+# UPBDEFS_GENERATOR - The upbdefs plugin executable (full path if not found in ${PATH})
+#
+# =============================================================================
+# Copyright 2009 Kitware, Inc. Copyright 2009-2011 Philip Lowman
```

Comment:
Google policy requires that all code that is not Copyright Google must go in the `third_party` directory.